lib/ui/volatile_path_tracker.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/volatile_path_tracker.h"
+
+namespace flutter {
+
+VolatilePathTracker::VolatilePathTracker(
+    fml::RefPtr<fml::TaskRunner> ui_task_runner)
+    : ui_task_runner_(ui_task_runner) {}
+
+void VolatilePathTracker::insert(std::shared_ptr<Path> path) {
+  FML_DCHECK(ui_task_runner_->RunsTasksOnCurrentThread());
+  FML_DCHECK(path);
+  FML_DCHECK(path->path_.isVolatile());
+  paths_.insert(path);
+}
+
+void VolatilePathTracker::erase(std::shared_ptr<Path> path) {
+  FML_DCHECK(path);
+  fml::TaskRunner::RunNowOrPostTask(ui_task_runner_,
+                                    [&]() { paths_.erase(path); });
+}
+
+void VolatilePathTracker::onFrame() {
```

Comment:
It might be nice to add some tracing here. For example, how many paths are tracked, how many survived, and how many are marked as non-volatile.

Reply:
Added some arguments.